examples/ble-gatt-client/src/main/java/no/nordicsemi/android/ble/ble_gatt_client/GattService.kt

```diff
@@ -0,0 +1,267 @@
+        if (bluetoothManager.adapter?.isEnabled == true) {
+            Log.i(TAG, "Enabling BLE services")
+            bluetoothManager.adapter.bondedDevices.forEach { device -> addDevice(device) }
+        } else {
+            Log.w(
+                    TAG,
+                    "Cannot enable BLE services as either there is no Bluetooth adapter or it is disabled"
+            )
+        }
+    }
+
+    private fun disableBleServices() {
+        clientManagers.values.forEach { clientManager ->
+            clientManager.close()
+        }
+        clientManagers.clear()
+    }
+
+    private fun addDevice(device: BluetoothDevice) {
+        if (!clientManagers.containsKey(device.address)) {
+            val clientManager = ClientManager()
+            clientManager.connect(device).useAutoConnect(true).enqueue()
+            clientManagers[device.address] = clientManager
+        }
+    }
+
+    private fun removeDevice(device: BluetoothDevice) {
+        clientManagers.remove(device.address)?.close()
+    }
+
+    /*
+     * Manages the entire GATT service, declaring the services and characteristics on offer
+     */
+    companion object {
+        /**
+         * A binding action to return a binding that can be used in relation to the service's data
+         */
+        const val DATA_PLANE_ACTION = "data-plane"
+
+        private const val TAG = "gatt-service"
+    }
+
+    private inner class ClientManager : BleManager(this@GattService) {
+        override fun getGattCallback(): BleManagerGattCallback =
+                GattCallback()
+
+        override fun log(priority: Int, message: String) {
+            if (BuildConfig.DEBUG || priority == Log.ERROR) {
+                Log.println(priority, TAG, message)
+            }
+        }
+
+        private inner class GattCallback : BleManagerGattCallback() {
+
+            private var myCharacteristic: BluetoothGattCharacteristic? = null
+
+            override fun isRequiredServiceSupported(gatt: BluetoothGatt): Boolean {
+                val service = gatt.getService(MyServiceProfile.MY_SERVICE_UUID)
+                myCharacteristic =
+                        service?.getCharacteristic(MyServiceProfile.MY_CHARACTERISTIC_UUID)
+                val myCharacteristicProperties = myCharacteristic?.properties ?: 0
+                return (myCharacteristicProperties and BluetoothGattCharacteristic.PROPERTY_READ != 0) &&
+                        (myCharacteristicProperties and BluetoothGattCharacteristic.PROPERTY_NOTIFY != 0)
+            }
+
+            override fun initialize() {
+                setNotificationCallback(myCharacteristic).with { _, data ->
+                    if (data.value != null) {
+                        val value = String(data.value!!, Charsets.UTF_8)
+                        defaultScope.launch {
+                            myCharacteristicChangedChannel?.send(value)
+                        }
+                    }
+                }
```

Comment:
Note that I had to move this callback outside of the queue statement below, where I previously used a `with` on `enableNotifications`. Could the fact that I had to do this be a bug?